### PR TITLE
Fixes #302 Fix tests in AndroidLibrary PR#214 and resubmit PR

### PR DESCRIPTION
**Pull request by WaltonG**
Fixes #302
-Handled null pointer exception and deprecated library for the tests to pass

### CODE REVIEW

**Comment by wbrunette** [comment]
@WaltonG. Before code is released we test it on the "development" branch. After changing the code to merge to development there is a merge problem. Can you fix? Thanks.

**Comment by WaltonG** [comment]
@wbrunette The fix has been made

**Comment by wbrunette** [comment]
@WaltonG note the PR still shows a conflict. The system won't let things move forward until it is resolved.
PRScreenshot

**Comment by WaltonG** [comment]
@wbrunette My thought is issue #302 is to fix tests in AndroidLibary and maintain the code. "We will need the code that was in PR#214 but we need the Android unit tests to pass."

The unit tests have passed. The merge conflict is caused by the existing code in PR#214 . Changes made to the PropertiesNonPrivilegedTest.java to have the unit test pass do not have any merge conflict

**Comment by wbrunette** [comment]
@WaltonG This is part of open-source as multiple people are making changes to improve things. If something becomes out of date it needs to be brought up to date with the current state of things. That is why GitHub will not let me accept your contribution until it works with all the code. Code is constantly changing which means that to complete a task sometimes you have to bring in other changes that have happened. This is standard procedure for software engineers.

**Comment by WaltonG** [comment]
@wbrunette Well understood. I'll work on the merge conflict right away

**Comment by WaltonG** [comment]
@wbrunette Retry the merging to confirm resolution of the conflict

**Comment by wbrunette** [comment]
@WaltonG the PR tells you if you can merge. Notice the screenshot. It is simply this same webpage.

**Comment by wbrunette** [comment]
@WaltonG I know this is new to you so don't worry about taking a while. You are here to learn. However, I want you to do it the right way so you learn.

File: androidlibrary_lib/src/main/java/org/opendatakit/properties/PropertiesSingleton.java [hunk 1]
```diff
@@ -488,6 +488,7 @@ private void init() {
     // and update them from the mDeviceDefaults map.
     for (Map.Entry<String, String> entry : mDeviceDefaults.entrySet()) {
       if (!mDeviceProps.containsKey(entry.getKey())) {
+        if(entry.getKey() != null && entry.getValue() != null)
```
Comment (wbrunette) [inline androidlibrary_lib/src/main/java/org/opendatakit/properties/PropertiesSingleton.java:491]: You should have brackets around 491, 492 and 493. As you don't want to set the updatedDeviceProps flag to true unless you actually set a property.

**Review by wbrunette — CHANGES_REQUESTED** [review]

**Comment by wbrunette** [comment]
@WaltonG I fixed the merge conflicts check it out.

**Review by WaltonG — COMMENTED** [review]
@wbrunette The changes fixes it right

**Comment by wbrunette** [comment]
@WaltonG while you changes do fix it, you have a logic error that I was trying to explain. If the "if" statement you put in fails, so no changes happen you still set the updated properties flag. However, you did not update the properties. Hence I requested changes and pointed out the line numbers.

**Comment by WaltonG** [comment]
@wbrunette I have pushed the committed changes

**Review by wbrunette — APPROVED** [review]

**Pull request merged**